pkg/syncer/broker/syncer_test.go

```diff
@@ -267,6 +322,89 @@ var _ = Describe("Broker Syncer", func() {
 		})
 	})
 
+	When("a synced non-local resource is stale in the local datastore on startup", func() {
+		BeforeEach(func() {
+			resource.SetNamespace(config.LocalNamespace)
+			test.SetClusterIDLabel(resource, "remote")
+			initialLocalResources = append(initialLocalResources, resource)
+		})
+
+		It("should delete it from the local datastore on reconciliation", func() {
+			test.AwaitNoResource(localClient, resource.GetName())
+		})
+	})
+
+	When("a synced non-local resource exists locally and in the broker datastore on startup", func() {
+		BeforeEach(func() {
+			resource.SetNamespace(config.LocalNamespace)
+			test.SetClusterIDLabel(resource, "remote")
+			initialLocalResources = append(initialLocalResources, resource)
+
+			copy := resource.DeepCopy()
+			copy.SetNamespace(config.BrokerNamespace)
+			initialBrokerResources = append(initialBrokerResources, copy)
+		})
+
+		It("should not delete it from the local datastore on reconciliation", func() {
+			time.Sleep(100 * time.Millisecond)
```

Comment:
I was thinking if there would be a better way for doing this, may be we can put all the sleep values under a constant at least, if we needed to bump them we could bump them all at once.

Reply:
This sleep isn't sensitive in that too small an interval might cause intermittent failures. I do this to give the code time to process it, such that if there's a bug and it does delete it, we can detect it. Otherwise, it may still exist b/c the code in question hasn't  executed yet.

Reply:
True, I was wondering if we could wait on the reconcile process to have finished  (at that point we can check that nothing was deleted). But it's ok for now I think

Reply:
yeah we'd have to somehow provide a chan for the tests to wait on. But even that wouldn't be complete as the delete is enqueued and processed on the workqueue. Instead, we could start an informer and wait for a delete update.